src/IonTimestamp.ts

```diff
@@ -185,6 +185,27 @@ export class Timestamp {
     );
   }
 
+  /**
+   * Static factory method for creating a Timestamp from a JavaScript Date object
+   *
+   * @param date - valid JavaScript date object
+   */
+  static fromDate(date: Date): Timestamp {
+    const seconds = new Decimal(
```

Comment:
Capturing the `Date`'s seconds as a decimal to preserve millisecond precision is the right idea here, but `Decimal`'s constructor is expecting an coefficient/exponent pair (scientific notation components) rather than an integer/fraction pair. You can find the documentation [here](https://github.com/amzn/ion-js/blob/3554f29b438fbc7021cf3760a46775257514be1d/src/IonDecimal.ts#L21-L37).